generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -823,19 +823,40 @@ public Expr apply(Expr expr, BasicType type, boolean constContext, int depth,
       assert type.isVector() || type.isMatrix();
       assert expr instanceof VariableIdentifierExpr;
 
-      final int numIndices =
+      final int numColumns =
           (BasicType.allVectorTypes().contains(type)
           ? type.getNumElements() : type.getNumColumns());
-      final int indexToFurtherTransform = generator.nextInt(numIndices);
+      final int columnToFurtherTransform = generator.nextInt(numColumns);
       final List<Expr> typeConstructorArguments = new ArrayList<>();
-      for (int i = 0; i < numIndices; i++) {
-        Expr argument = new ArrayIndexExpr(expr.clone(), new IntConstantExpr(String.valueOf(i)));
-        if (i == indexToFurtherTransform) {
-          argument = applyIdentityFunction(argument,
-              (type.isVector() ? type.getElementType() : type.getColumnType()),
-              constContext, depth, fuzzer);
+
+      if (type.isMatrix() && generator.nextBoolean()) {
+        // further break down matrix constructor into single components, to increase diversity
```

Comment:
Elaborate the last sentence of the comment more. "For example, a mat2x2 m might turn into: ..."